### PR TITLE
Add try_insert/insert, basic insert test, bump version

### PR DESCRIPTION
**Pull request by hazelutf8**
When using it seemed like an insert function could be useful.

Please advise if any changes are needed, or if only a subset of these changes are wanted.

### CODE REVIEW

**Comment by ZackPierce** [comment]
Thanks, @hazelutf8 , I agree that `try_insert` and `insert` seem like useful functions to have.

I would prefer a little bit more testing before this gets merged into main. 

First, it would be nice to broaden the unit testing to concretely demonstrate the error-cases for `try_insert` (insufficient capacity, index out of bounds), and the cases where one inserts at a position other than `0`.

Second, for the comparability tests, even though there is a branch for handling `VecLikeOp::Insert`, the strategy that creates the `VecLikeOp` instances, `arbitrary_vec_like_op`, does not yet create the `Insert` variant, and ought to.  I suspect that once we start making such instances and run the tests, we will discover that a few aspects of that test branch ought to change.


If you have the time and willingness to expand the testing, please go ahead. If not, drop a note and I will add said tests when I get the chance.

Thanks again for the contribution!

**Comment by hazelutf8** [comment]
Let me know if any `proptest` related items are missing since I am new to its use, or if there are any test cases I may have forgotten.

File: tests/comparability_tests.rs [hunk 1]
```diff
@@ -481,10 +482,14 @@ pub mod vec_like_operations {
                 VecLikeOp::Insert(i, v) => {
                     let fs_result = fs_vec.try_insert(i, v);
                     if let Err(e) = fs_result {
-                        prop_assert!(fs_vec.is_full(), "FixedSliceVec should only reject inserts when full. Failed pushing {:?}", e);
+                        let insert_failure_expected =
+                            (other_vec.capacity() == other_vec.len()) || (i > other_vec.len());
+                        prop_assert!(insert_failure_expected, "FixedSliceVec should only reject when full or insertion index outside valid range. Failed inserting {:?} at {:?}", e, i);
                     } else {
                         if let Err(e) = other_vec.try_push(v) {
```
Comment (ZackPierce) [inline tests/comparability_tests.rs]: This should be `try_insert` in order to parallel correctly.

**Comment by hazelutf8** [comment]
Closing in favor of #21, to reduce unrelated changes.

**Pull request closed without merging**